### PR TITLE
Best practices document

### PR DESCRIPTION
**Pull request by Nerdy-Girl**
Issue: #27

### CODE REVIEW

File: BEST_PRACTICES.md [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+  .title {
+    ...
+  }
+}
+```
+
+Bad practice:
+```scss
+.container {
+     @include pl(2);
```
Comment (ArekRado) [inline BEST_PRACTICES.md]: it should be real tab

File: BEST_PRACTICES.md [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+### Global styles
+
+Sometimes we find a package with pretty awesome feature we want to add in our project but it doesn't fit our design. 
+To solve this problem good practice is to overwrite feature's styles with our own but in React it may be very tricky 
+(especially while using CSS Modules). That's why we recommend overwriting them in `:global` class. 
+
+Thanks to that during build step given class will be clear without any dynamically generated, unique, and mapped to the 
+correct styles additional text. They won't be scoped to particular templates but will be accessible from global scope.
+
+Good practice:
```
Comment (ArekRado) [inline BEST_PRACTICES.md]: plz add more examples, how to style external react-components which don't use css-modules: (it's good practice to wrap each `global` by normal classes)

```scss
.cell {
  @include flex($direction: column);
  @include p(0.5, 1);
  @include m(0.5, 0);

  :global {
    .Select.is-focused > .Select-control,
    .Select-control {
      border: none;
      border-bottom: 1px solid $color-gray-darker;
      background-color: transparent;
      border-radius: 0;
    }
  }
}
```

Comment (ArekRado) [reply BEST_PRACTICES.md]: how to style another react components which use css-modules, is it possible to change theirs styles?

**Review by ArekRado — CHANGES_REQUESTED** [review]

File: BEST_PRACTICES.md [hunk 3]
```diff
@@ -0,0 +1,549 @@
...
+
+As you already should know we like to keep our code base clear and transparent. That's the main reason why we keep order while declaring classes and their states. 
+
+We follow the rule that first state of current class should be declared, then we declare nested children. Thanks to 
+that we can easily see what properties given class can accept before focusing on child classes. Doing it in reverse 
+order can mislead you while reading code.
+
+Good practice:
+```scss
+.container {
```
Comment (JSerhii) [inline BEST_PRACTICES.md:288]:
```css
.container {
   &.active {
     ...
   }
}
```

File: BEST_PRACTICES.md [hunk 4]
```diff
@@ -0,0 +1,549 @@
...
+  .active {
+    ...
+  }
+  .hovered {
+    ...
+  }
+  
+  .button {
```
Comment (JSerhii) [inline BEST_PRACTICES.md]:
```
.button {
   &.hovered {
     ...
   }
}
```

File: BEST_PRACTICES.md [hunk 5]
```diff
@@ -0,0 +1,549 @@
...
+In styling there are two opposite theories about using mixin in specified order within class. First one says
+that firstly you should use mixin and then declare you own styles. Second one claims that you should do it in opposite order.
+
+We practice the first one and we believe that you should too. It will make your work more clear and lucid. Thanks to that solution
+you can easily override mixins styles and while maintaining it will help you understand styles order faster. 
+
+Good practice:
+```scss
+.container {
+  @include p(2, 2, 2, 2);
```
Comment (JSerhii) [inline BEST_PRACTICES.md]: add another example
```
@include p(2, 3, 2, 4);
```

or 

```
@include p(2);
```

**Review by JSerhii — APPROVED** [review]

**Review by ArekRado — APPROVED** [review]

**Pull request merged**